scripts/gitlab/publish-snap.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e # fail on any error
+set -u # treat unset variables as error
+
+case ${CI_COMMIT_REF_NAME} in
+  nightly|*v2.3*) export GRADE="devel" CHANNEL="edge";;
+  beta|*v2.2*) export GRADE="stable" CHANNEL="beta";;
+  stable|*v2.1*) export GRADE="stable" CHANNEL="stable";;
```

Comment:
This is the only part that must to be updated when the repo changes.
Is there no way out of hard-coding there the v2.x version numbers?
If so, I recommend adding a comment explaining this, and ideally, a guide document with the things to do when re-targetting beta and stable.

Reply:
Yes, we have this internal documentation for releases. I honestly have no better idea how to map `CI_COMMIT_REF_NAME` to `GRADE` and `CHANNEL`...

Reply:
My idea for automating the releases on Travis was to push any commit from the master branch to edge, from the beta branch to beta and from the stable branch to candidate.
But I don't know enough gitlab to understand when `CI_COMMIT_REF` is set to the tag and when to the branch name, and I couldn't find a variable that's always the branch name. 
If this hasn't been a problem in the past, there's no reason to fix it right now.

Reply:
> version and grade can be obtained with parsing `git branch --contains tags/$CI_COMMIT_REF_NAME`or sort of.

nice idea, works only partially that way (e.g. git branch --contains tags/v2.2.1 doesn't)